api/cluster.go

```diff
@@ -353,14 +354,16 @@ func (s *Server) peerQuery(ctx context.Context, data cluster.Traceable, name, pa
 	}()
 
 	result := make(map[string]PeerResponse)
+	errors := make(map[string]error)
 	for resp := range responses {
 		if resp.err != nil {
-			return nil, resp.err
```

Comment:
> but with this change, this is no longer true

Why is this no longer true? The request context's `cancel` still gets called, which makes `peer.Post` not wait for the response.

The change should only make this function return more details regarding which peers failed with what errors, as opposed to just returning "there was some error". This can be useful for example if the upsert of a meta record fails and the user would like to know which peers already executed that operation successfully and which didn't.
In the case of an error (or multiple), this change might make the function return a bit slower than before, because it still needs to collect all the errors and responses. 
In the case of no error the execution time should be the same.

Reply:
yep i was wrong.
ok makes sense